### PR TITLE
Add e2e test for 'Basic functionality' part in demo

### PR DESCRIPTION
**Pull request by GinnyJI**
Tested: make e2e-test

### CODE REVIEW

File: incubator/hnc/test/e2e/demo_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "sigs.k8s.io/multi-tenancy/incubator/hnc/pkg"
+)
+
+var _ = Describe("Subnamespaces", func() {
```
Comment (adrianludwin) [inline incubator/hnc/test/e2e/demo_test.go:8]: s/Subnamespaces/Demo/

Comment (adrianludwin) [reply incubator/hnc/test/e2e/demo_test.go:8]: Also add a comment pointing to the demo doc, and (once this is approved) add a warning to the top of the demo doc asking authors to keep it in sync with this file.

File: incubator/hnc/test/e2e/demo_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		MustNotRun("kubectl hns set", nsOrg, "--parent", nsTeamA)
+		// verify the tree
+		RunShouldContain(nsTeamA, 5, "kubectl hns describe", nsOrg)
+		RunShouldContain(nsService1, 5, "kubectl hns describe", nsTeamA)
+
+		// Now, if we check service-1 again, we’ll see all the rolebindings we expect:
+		RunShouldContainMultiple([]string{"hnc.x-k8s.io/inheritedFrom=acme-org", "hnc.x-k8s.io/inheritedFrom=team-a"},
+			 5, "kubectl -n", nsService1, "describe roles")
+		RunShouldContainMultiple([]string{"Role/acme-org-sre", "Role/team-a-sre"}, 5, "kubectl -n", nsService1, "get rolebindings")
+		// TODO: check output
```
Comment (adrianludwin) [inline incubator/hnc/test/e2e/demo_test.go:48]: Fix?

Comment (GinnyJI) [reply incubator/hnc/test/e2e/demo_test.go:48]: sry forgot to remove..

File: incubator/hnc/test/e2e/demo_test.go (hunk 1 above)
Comment (GinnyJI) [reply incubator/hnc/test/e2e/demo_test.go]: done

**Pull request merged**